### ISSUE TITLE
Toggleterm and Lazygit ESC does not work

### ISSUE THREAD

**Issue by saidbelkacem** [AstroNvim/AstroNvim#627]
Hello friendly users. Absolutely love AstroNvim :-)

Using Lazygit with <leader>gg and doing something like SHIFT ? for help, then help pops up as expected but I get stuck and cant get out with ESC. Any idea why?

**Comment by mehalter** [comment]
Duplicate of #356

**Comment by mehalter** [comment]
We have some default terminal bindings in place to make it easier to leave insert mode in the terminal. You can delete these default bindings in your `polish` function and use the default terminal bindings or remap them to something that doesn't conflict with escape

**Comment by saidbelkacem** [comment]
Thanks Micah
But I am wondering how do you personally get out in this specific case in Lazygit? I dont get the idea behind this behaviour.

**Comment by mehalter** [comment]
I remap these bindings in my user configuration to `ctrl-q` to leave insert mode and then `<esc><esc>` to quickly close the terminal: 

```lua
local map = vim.keymap.set
local unmap = vim.keymap.del

unmap("t", "<esc>")
unmap("t", "jk")

map("t", "<c-q>", "<c-\\><c-n>", { desc = "Terminal normal mode" })
map("t", "<esc><esc>", "<c-\\><c-n>:q<cr>", { desc = "Terminal quit" })
```

I personally unmap a lot of default AstroNvim bindings that I disagree with and don't really fit well into my personal workflow.

**Comment by Noxdor** [comment]
Hey, I just stumbled across this and wanted to give my opinion on the default setting:

I can fully understand that people who are vim enthusiasts want the esc behaviour in the terminal too. But I don't understand this choice for this project and specifically for the lazygit integration.

Imo, the great thing about AstroNvim is, that it just works. Without knowing too much about Nvim, I can start working, it is a well orchestrated system of modules. Seeing this, I can't understand why the default setting would break something as essential as the back button in lazygit, which is a core integration of AstroNvim. Without changing the default key-map, I have to fully close AstroNvim to stop an action in lazygit, that I started wrongly.
Since vim enthusiasts are already familiar with how to change key-maps etc., why not let the "easy" and in my opinion "working" way as the default key-map and let the enthusiast set, how he or she want their keys to work? This would make it even easier for beginner to join the Nvim world, draw from the benefits of this awesome application and get going fast with AstroNvim!

Anyway, just my opinion as an AstroNvim user now since a good few months. Cheers!
